rabbitmq/src/main/kotlin/RabbitMQMessageAdapter.kt

```diff
@@ -0,0 +1,93 @@
+import com.rabbitmq.client.AMQP
+import com.rabbitmq.client.Channel
+import com.rabbitmq.client.ConnectionFactory
+import io.github.positionpal.AddedMemberToGroup
+import io.github.positionpal.Event
+import io.github.positionpal.EventSerializer
+import io.github.positionpal.GroupCreated
+import io.github.positionpal.GroupDeleted
+import io.github.positionpal.MessageType
+import io.github.positionpal.RemovedMemberToGroup
+
+/**
+ * A message adapter for RabbitMQ to handle posting events and managing connections.
+ *
+ * @property rabbitMQHost The RabbitMQ host address.
+ * @property rabbitMQPort The RabbitMQ port number.
+ * @property rabbitMQUsername The RabbitMQ username.
+ * @property rabbitMQPassword The RabbitMQ password.
+ * @property serializer The serializer for events.
+ */
+class RabbitMQMessageAdapter(
+    private val rabbitMQHost: String = "localhost",
+    private val rabbitMQPort: Int = 5672,
+    private val rabbitMQUsername: String = "guest",
+    private val rabbitMQPassword: String = "guest",
+    private val serializer: EventSerializer,
+) : MessageAdapter {
+    private val factory = ConnectionFactory().apply {
+        host = rabbitMQHost
+        port = rabbitMQPort
+        username = rabbitMQUsername
+        password = rabbitMQPassword
+    }
+    private val connection = factory.newConnection()
+    private val channel: Channel = connection.createChannel()
+
+    init {
+        channel.exchangeDeclare("group_updates_exchange", "headers")
+        channel.queueDeclare("group_updates", false, false, false, null)
+
+        val args = mapOf(
+            "x-match" to "all",
+            "message_type" to "update",
+        )
+        channel.queueBind("group_updates", "group_updates_exchange", "", args)
+    }
+
+    /**
+     * Posts an event to the RabbitMQ exchange.
+     *
+     * @param type The type of the message.
+     * @param event The event to be posted.
+     * @throws IllegalArgumentException if the event type is not supported.
+     */
+    override fun postEvent(type: MessageType, event: Event) {
+        val headers = mapOf(
+            "message_type" to type.toString(),
+        )
+
+        val properties = AMQP.BasicProperties.Builder()
+            .headers(headers)
+            .build()
+
+        val byte = when (type) {
+            MessageType.GROUP_CREATED ->
+                serializer.serializeGroupCreated(event as GroupCreated?)
+            MessageType.GROUP_DELETED ->
+                serializer.serializeGroupDeleted(event as GroupDeleted?)
+            MessageType.USER_CREATED ->
+                throw IllegalArgumentException("User created event not supported")
+            MessageType.USER_DELETED ->
+                throw IllegalArgumentException("User deleted event not supported")
```

Comment:
If these types are not supported we can get rid of them in the shared kernel?

Reply:
yes, I can. I will proceed